src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/FileEntryCreator.kt

```diff
@@ -0,0 +1,54 @@
+package pl.edu.uj.ii.ksi.mordor.services
+
+import java.io.File
+import org.springframework.stereotype.Service
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileContent
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileEntry
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileMetadata
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileContentRepository
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileEntryRepository
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileMetadataRepository
+
+@Service
+class FileEntryCreator(
+    private val metadataExtractor: MetadataExtractor,
+    private val entryRepository: FileEntryRepository,
+    private val metadataRepository: FileMetadataRepository,
+    private val contentRepository: FileContentRepository,
+    private val fileTextExtractor: FileTextExtractor
+) {
+
+    fun create(file: File): FileEntry {
+        val metadata = getMetadata(file)
+        createContent(file, metadata)
+        return createEntry(file, metadata)
+    }
+
+    private fun createContent(file: File, metadata: FileMetadata) {
+        val text = fileTextExtractor.extract(file)!!
+        val content = FileContent(text = text, file = metadata)
+        contentRepository.save(content)
```

Comment:
Throws an error `detached entity passed to persist`.
Also... have you actually tested this before committing? Like whether it actually works?